### ISSUE TITLE
Is there any way to shorten the time of avroconverter?

### ISSUE THREAD

**Issue by bylee5** [Apicurio/apicurio-registry#1492]
hi, For large data, we are checking the json format and avro format, which are converter options in kafka.
The json format is too big for the data size, but the processing is fast.
The avro format has a small data size, but the processing is very slow.
The test results are as follows.
mysql table row: 167023, ibd: 1.1G
jsonconverter: 14 seconds, kafka topic size: 2.3G
avroconverter: 14min 08sec, kafka topic size: 1.1G
extjsonconverter: 14min 38sec, kafka topic size: 1.1G
Is there any way to shorten the time of avroconverter/extjsonconverter?

**Comment by EricWittmann** [comment]
I'm afraid I don't have any insight into this.  Any thoughts @famartinrh @gunnarmorling @jpechane @alesj  @Apicurio/developers ?

**Comment by jpechane** [comment]
@bylee5 Could you please share the connector configuration here together with the DDL of the table you were capturing?

**Comment by gunnarmorling** [comment]
Yes, with that info we should be able to do a profiling with JFR and see
where any slownesses are.

>

**Comment by bylee5** [comment]
> @bylee5 Could you please share the connector configuration here together with the DDL of the table you were capturing?
DDL and source connector information is as follows.
DDL:
CREATE TABLE `deal_check_list_repo` (
  `deal_check_list_repo_seqno` int(11) NOT NULL AUTO_INCREMENT,
  `main_deal_no` bigint(20) NOT NULL,
  `chklst_group` varchar(45) NOT NULL ,
  `chklst_data` mediumtext NOT NULL ,
  `create_dt` datetime NOT NULL ,
  `creator` varchar(50) NOT NULL,
  `update_dt` datetime NOT NULL DEFAULT,
  `updater` varchar(50) DEFAULT,
  PRIMARY KEY (`deal_check_list_repo_seqno`),
  UNIQUE KEY `idx_main_deal_no` (`main_deal_no`)
) ENGINE=InnoDB  DEFAULT CHARSET=utf8 ROW_FORMAT=COMPACT;

source connector:
curl -i -X POST -H "Accept:application/json" \
    -H "Content-Type:application/json" http://localhost:8083/connectors/ \
    -d '{
                "name": "source-mysql-avro-buy-4",
                "config": {
                    "connector.class": "io.debezium.connector.mysql.MySqlConnector",
                    "tasks.max": "1",
                    "time.precision.mode": "connect",
                    "key.converter": "io.apicurio.registry.utils.converter.AvroConverter",
					"key.converter.apicurio.registry.url": "http://localhost:8080/api",
					"key.converter.apicurio.registry.global-id": "io.apicurio.registry.utils.serde.strategy.GetOrCreateIdStrategy",
					"value.converter": "io.apicurio.registry.utils.converter.AvroConverter",
					"value.converter.apicurio.registry.url": "http://localhost:8080/api",
					"value.converter.apicurio.registry.global-id": "io.apicurio.registry.utils.serde.strategy.GetOrCreateIdStrategy",
                    "database.hostname": "upddb2.tmonc.net",
                    "database.port": "3306",
                    "database.user": "debezium",
                    "database.password": "adm@)!*84",
                    "database.server.id": "205011",
                    "database.server.name": "dbserver4",
                    "database.history.kafka.bootstrap.servers": "localhost:9092",
                    "database.history.kafka.topic": "schema-changes.inventory",
                    "include.schema.changes": "true",
                    "database.include.list": "inventory",
                    "table.include.list": "inventory.deal_check_list_repo",
                    "database.serverTimezone": "Asia/Seoul",
                    "transforms": "route",
                    "transforms.route.type": "org.apache.kafka.connect.transforms.RegexRouter",
                    "transforms.route.regex": "([^.]+)\\.([^.]+)\\.([^.]+)",
                    "transforms.route.replacement": "avro-buy-4"
                }
            }'
